addons/survey/static/src/scss/survey_templates_results.scss

```diff
@@ -12,6 +12,37 @@
     html {
         height: unset;
     }
+    .o_survey_question_page {
+        page-break-inside: avoid;
+    }
+    .o_survey_results_question_wrapper {
+        .o_survey_results_question_header {
+            break-inside: avoid;
+            page-break-after: avoid;
+        }
+        .o_survey_question_description {
+            break-inside:avoid;
+            page-break-before: avoid;
+        }
+    }
+    table {
+        thead {
+            display: table-row-group;
+        }
+        tbody {
+            tr {
+              break-inside: avoid;  
+            }
+        }
+    }
```

Comment:
These rules are here to avoid some element like table row, image... being split into two pages. I am replacing the `break-*` by `page-break-*` rules as they are specific to print mode. `t-head { display: table-row-group } `is here to avoid repeating the table header when a table spreads over 2 or more pages.